enhancements/20240930-vxlan-enhancement.md

```diff
@@ -0,0 +1,197 @@
+
+(2) Create one VxLAN overlay network on top of the guest Cluster Network, say VxLAN_1, VNI 42, the overlay network CIDR, e.g. 10.42.0.0/8. The default port number can be configured manually or automatically. The port 8472 is occupied for `flannel` by default. The port 4789 is used for demonstration in the following operations. The Local IP and remote peers of the VTEP in each NODE are fetched from APIServer. The vxlan interface will be created with the specified VNI on the NODE.
+```
+ip link add vxlan42 type vxlan id 42 dstport 4789 local 192.168.5.11 dev cn-vxlan_1-bo
+ip link set vxlan42 master cn-vxlan_1-br
+```
+(3) Create VMs attached to the VxLAN_1, and move one of the VM to NODE_2/NODE_3.
+The bridge/bonding interfaces of the guest Cluster Network should be created automatically on NODE_2/NODE_3. 
+The bonding interface will play as the VTEP which needs the IP address to be assigned automatically from DHCP.
+The `vxlan42` interface should also be created and attached to the brige interface like (2). 
+```
+on NODE_2
+ip link add vxlan42 type vxlan id 42 dstport 4789 local 192.168.5.22 dev cn-vxlan_1-bo
+ip link set vxlan42 master cn-vxlan_1_br
+
+on NODE_3
+ip link add vxlan42 type vxlan id 42 dstport 4789 local 192.168.5.33 dev cn-vxlan_1-bo
+ip link set vxlan42 master cn-vxlan_1-br
+```
+(4) Note that we don't use `group` option while creating the vxlan interface, the broadcast packets will not be sent to the multicast group to learn the FDB entry from other VTEPs.
+We need to manually add the FDB entry on each NODE so the VM on the VxLAN network can find each other by ARP. 
+```
+on Mgmt NODE:	add NODE_2, NODE_3 in the forwarding table. The broadcast packets would be correctly forwarded and the gratuitous ARP can work on top of it.
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.22
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.33
+bridge fdb append 02:22:22:22:22:22 dev vxlan42 dst 192.168.5.22
+bridge fdb append 02:33:33:33:33:33 dev vxlan42 dst 192.168.5.33
+
+on NODE_2:	add Mgmt NODE and NODE_3 in the forwarding table.
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.11
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.33
+bridge fdb append 02:11:11:11:11:11 dev vxlan42 dst 192.168.5.11
+bridge fdb append 02:33:33:33:33:33 dev vxlan42 dst 192.168.5.33
+
+on NODE_3:	add Mgmt NODE and NODE_2 
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.11
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.22
+bridge fdb append 02:11:11:11:11:11 dev vxlan42 dst 192.168.5.11
+bridge fdb append 02:22:22:22:22:22 dev vxlan42 dst 192.168.5.22
+```
+
+(5) The VxLAN interface on each NODE should play as the ARP proxy for VMs attached to the same VxLAN network.
+Then the packet destined to the VM have the ARP response from the corresponding VTEP and correctly forwarded.
+Adding the ARP entry for each VM for other NODEs which has no information for it. 
```

Comment:
I didn't get this.
Your `vxlan0` interface will be bridge slave in this setup.
The ARP entry will not be used in vxlan0 interface.
If you want "arp suppression" feature, you will need to assign IP address on every vxlan bridge and enable "arp suppression".
Your bridge will learn ARP from ARP response and record in bridge's ARP table.
If your bridge have ARP entry, it will reply the ARP request directly next time.
But in most of time, if you didn't use BGP-EVPN, I will not suggest to enable any ARP suppression feature.